python/ray/worker.py

```diff
@@ -2350,21 +2364,25 @@ def register_custom_serializer(cls,
         # and not across workers.
         class_id = random_string()
 
+    if driver_hash == None:
+        driver_hash = worker.task_driver_id.__hash__()
+
     def register_class_for_serialization(worker_info):
         # TODO(rkn): We need to be more thoughtful about what to do if custom
         # serializers have already been registered for class_id. In some cases,
         # we may want to use the last user-defined serializers and ignore
         # subsequent calls to register_custom_serializer that were made by the
         # system.
-        worker_info["worker"].serialization_context.register_type(
+        serialization_context = worker_info["worker"].get_serialization_context(driver_hash)
```

Comment:
Instead of using a hash, we can define

```python
driver_id_bytes = worker.task_driver_id.id()
```

and then here we can do `get_serialization_context(ray.ObjectID(driver_id_bytes))`.

Then we can get rid of all of the calls to `__hash__()`.

Reply:
Later we can just make `ObjectID`s pickleable (which should be pretty simple).

Reply:
ok